numpy/core/src/multiarray/buffer.c

```diff
@@ -312,8 +314,13 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
 
         *offset += descr->elsize;
 
-        if (descr->byteorder == '=' &&
-                _is_natively_aligned_at(descr, arr, *offset)) {
+        if(!PyArray_IsScalar(obj, Generic)) {
+            /* only check ndarrays, scalars are always natively aligned */
```

Comment:
This would be clearer if you assigned the `is_natively_aligne = 1` in the `else`, and moved the comment there (or swapped the `if` and `else`